{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "167c2961-c88c-41e1-a44a-21bf349abacd",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(dplyr)\n",
    "library(tidyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "fd45fb6b-45a4-4ffc-8d0d-add40bef08b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "All_RBP_motifs <- read.csv(\"/home/ylee/AllmotifDB/All_RBP_motifs.csv\")\n",
    "rbp_list <- unique(All_RBP_motifs$RBP_name)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0874c7ea-318d-4d25-a559-f4f46730cc04",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "## Function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "b5240734-8461-4d79-a0c7-c0b10dc8f21b",
   "metadata": {},
   "outputs": [],
   "source": [
    "## alignment_details functions\n",
    "\n",
    "alignment_details <- function(rbp_list, RBP_motifs, fimo_base_path) {\n",
    "    motif_details <- data.frame(\n",
    "        database = character(),\n",
    "        RBP_name = character(),\n",
    "        motif_id = character(),\n",
    "        motif_alt_id = character(),\n",
    "        alignment = logical(),\n",
    "        start = numeric(),\n",
    "        stop = numeric(),\n",
    "        strand = character(),\n",
    "        score = numeric(),\n",
    "        p_value = numeric(),\n",
    "        q_value = numeric(),\n",
    "        matched_sequence = character(),\n",
    "        stringsAsFactors = FALSE\n",
    "    )\n",
    "    \n",
    "    for (rbp in rbp_list){\n",
    "        fimo_file <- paste0(fimo_base_path, rbp, \"/fimo.tsv\")\n",
    "        fimo_data <- read.delim(fimo_file, header = TRUE)\n",
    "\n",
    "        motif_data <- RBP_motifs %>%\n",
    "            filter(RBP_name == rbp) %>%\n",
    "            select(database, RBP_name, motif_id, motif_alt_id)\n",
    "\n",
    "        final_data <- list()\n",
    "        for (i in 1:nrow(motif_data)) {\n",
    "            if (\"motif_id\" %in% colnames(fimo_data) && \"motif_alt_id\" %in% colnames(fimo_data)){\n",
    "                fimo_row <- fimo_data %>%  \n",
    "                    filter(motif_id == motif_data[i, \"motif_id\"], motif_alt_id == motif_data[i, \"motif_alt_id\"])\n",
    "                \n",
    "                if (nrow(fimo_row) > 0) {\n",
    "                    # If alignment data exists, fill in alignment data\n",
    "                    alignment <- TRUE\n",
    "                    start <- fimo_row$start\n",
    "                    stop <- fimo_row$stop\n",
    "                    strand <- fimo_row$strand\n",
    "                    score <- fimo_row$score\n",
    "                    p_value <- fimo_row$p.value\n",
    "                    q_value <- fimo_row$q.value\n",
    "                    matched_sequence <- fimo_row$matched_sequence\n",
    "                } else {\n",
    "                    # If no alignment data, set to NA\n",
    "                    alignment <- FALSE\n",
    "                    start <- NA\n",
    "                    stop <- NA\n",
    "                    strand <- NA\n",
    "                    score <- NA\n",
    "                    p_value <- NA\n",
    "                    q_value <- NA\n",
    "                    matched_sequence <- NA\n",
    "                }\n",
    "            }else {\n",
    "                    # If no alignment data, set to NA\n",
    "                    alignment <- FALSE\n",
    "                    start <- NA\n",
    "                    stop <- NA\n",
    "                    strand <- NA\n",
    "                    score <- NA\n",
    "                    p_value <- NA\n",
    "                    q_value <- NA\n",
    "                    matched_sequence <- NA\n",
    "            }\n",
    "\n",
    "            \n",
    "            # Create a data frame for the current motif and append to motif_details\n",
    "            motif_details <- bind_rows(motif_details, data.frame(\n",
    "                database = motif_data[i, \"database\"],\n",
    "                RBP_name = motif_data[i, \"RBP_name\"],\n",
    "                motif_id = motif_data[i, \"motif_id\"],\n",
    "                motif_alt_id = motif_data[i, \"motif_alt_id\"],\n",
    "                alignment = alignment,\n",
    "                start = start,\n",
    "                stop = stop,\n",
    "                strand = strand,\n",
    "                score = score,\n",
    "                p_value = p_value,\n",
    "                q_value = q_value,\n",
    "                matched_sequence = matched_sequence,\n",
    "                stringsAsFactors = FALSE\n",
    "            ))\n",
    "        }\n",
    "    }\n",
    "    return(motif_details)    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "76789c74-841c-4598-b408-f72b7fec92a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "alignment_combined_details <- function(csv_paths, mapped_data, sequence_name) {\n",
    "    alignment_combined <- data.frame(\n",
    "        fragment_id = character(),\n",
    "        database = character(),\n",
    "        RBP_name = character(),\n",
    "        motif_id = character(),\n",
    "        motif_alt_id = character(),\n",
    "        alignment = logical(),\n",
    "        start = numeric(),\n",
    "        stop = numeric(),\n",
    "        strand = character(),\n",
    "        score = numeric(),\n",
    "        p_value = numeric(),\n",
    "        q_value = numeric(),\n",
    "        matched_sequence = character(),\n",
    "        stringsAsFactors = FALSE\n",
    "    )\n",
    "    \n",
    "    mapped_data[, 1] <- paste0(sequence_name, \"_\", mapped_data[, 1] + 1)\n",
    "    colnames(mapped_data)[1] <- \"fragment_id\"\n",
    "    mapped_data$motif_alignments <- NA\n",
    "    \n",
    "    for (csv_path in csv_paths){\n",
    "        fragment_number <- as.numeric(sub(paste0(\".*\", sequence_name, \"_(\\\\d+)\\\\.csv$\"), \"\\\\1\", csv_path))\n",
    "        fragment_start <- mapped_data[fragment_number, \"position.in.adapting.gene.DNA\"]\n",
    "        \n",
    "        alignment_df <- read.csv(csv_path)\n",
    "        filtered_df <- alignment_df[alignment_df$alignment == TRUE, ]\n",
    "        mapped_data$motif_alignments[fragment_number] <- nrow(filtered_df)\n",
    "        \n",
    "        if (nrow(filtered_df) > 0) {\n",
    "            filtered_df$fragment_id <- paste0(sequence_name, \"_\", fragment_number)\n",
    "            filtered_df$start <- filtered_df$start + fragment_start\n",
    "            filtered_df$stop <- filtered_df$stop + fragment_start\n",
    "            alignment_combined <- bind_rows(alignment_combined, filtered_df)\n",
    "        }\n",
    "    }\n",
    "\n",
    "    return(list(alignment_combined = alignment_combined, mapped_data = mapped_data))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16eb7626-97f7-4a6a-b946-fa6b25b8f4b6",
   "metadata": {},
   "source": [
    "## Combine Motif Scanning Result - Actg1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "78902272-cc57-4237-a877-819879a106a8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "sequence_name = \"Actg1_hotspot\"                            ### Change Here : Actg1_hotspot, Actg1_ctrl, Actg1_random\n",
    "base_dir <- paste0(\"/home/ylee/meme/fimo_out/\", sequence_name)\n",
    "csv_dir <- paste0(base_dir, \"/alignment_csv\")\n",
    "\n",
    "fimo_base_path <- paste0(base_dir, \"/\")\n",
    "dir.create(csv_dir)\n",
    "\n",
    "motif_details <- alignment_details(rbp_list, All_RBP_motifs, fimo_base_path)\n",
    "\n",
    "output_dir <- paste0(csv_dir, \"/\", basename(fimo_base_path), \".csv\")\n",
    "write.csv(motif_details, output_dir, row.names = FALSE) "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5b9a062-2294-4bfc-a410-c82871b684d3",
   "metadata": {},
   "source": [
    "## Combine Motif Scanning Result - FERMT1/TPM2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "177cbc4f-aff2-401f-a162-fd5bdd446206",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "sequence_name = \"FERMT1\"                                      ### Change Here : FERMT1, TPM2\n",
    "base_dir <- paste0(\"/home/ylee/meme/fimo_out/\", sequence_name)\n",
    "csv_dir <- paste0(base_dir, \"/alignment_csv\")\n",
    "\n",
    "fimo_base_paths <- paste0(list.dirs(base_dir, recursive = FALSE, full.names = TRUE), \"/\")\n",
    "dir.create(csv_dir)\n",
    "\n",
    "for (fimo_base_path in fimo_base_paths){\n",
    "    motif_details <- alignment_details(rbp_list, All_RBP_motifs, fimo_base_path)    \n",
    "    \n",
    "    output_dir <- paste0(csv_dir, \"/\", basename(fimo_base_path), \".csv\")\n",
    "    print(output_dir)\n",
    "    flush.console()\n",
    "    write.csv(motif_details, output_dir, row.names = FALSE)   \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "1c2fc4ff-de51-4dd1-a309-ec96a4bb530f",
   "metadata": {},
   "outputs": [],
   "source": [
    "if (sequence_name == \"FERMT1\"){\n",
    "    mapped_data <- read.csv(\"/home/ylee/meme/FERMT1_TPM2_sequence/PANX1_NB_fragment_FERMT1_mapped_sequecnes_64_percent.csv\", header = TRUE, fill = TRUE)\n",
    "} else {     \n",
    "    mapped_data <- read.csv(\"/home/ylee/meme/FERMT1_TPM2_sequence/PANX1_NB_fragment_TPM2_mapped_sequecnes_64_percent.csv\", header = TRUE, fill = TRUE)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "deb3bfee-3464-4991-a892-8585070ed638",
   "metadata": {},
   "outputs": [],
   "source": [
    "csv_paths <- list.files(csv_dir, recursive = FALSE, full.names = TRUE)\n",
    "csv_paths <- csv_paths[grepl(paste0(\"^\", sequence_name, \"_\\\\d+\\\\.csv$\"), basename(csv_paths))]\n",
    "\n",
    "results <- alignment_combined_details(csv_paths, mapped_data, sequence_name)\n",
    "write.csv(results$mapped_data, paste0(csv_dir, \"/\", sequence_name, \"_mapped_motif_alignments_summary.csv\"), row.names = FALSE) \n",
    "write.csv(results$alignment_combined, paste0(csv_dir, \"/\", sequence_name, \"_mapped_motif_alignments.csv\"), row.names = FALSE)  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d3feb72-7aea-48aa-a890-1fb8fccf9605",
   "metadata": {},
   "source": [
    "## Combine Motif Scanning Result-Actg2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "1a044664-30cf-44b4-bf8f-5cff2ef8fac3",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   database RBP_name motif_id motif_alt_id\n",
      "1 CisBP_RNA     FXR2     1.10      DGACRRR\n",
      "  motif_id motif_alt_id       sequence_name start stop strand  score p.value\n",
      "1     1.10      DGACRRR Actg2_ctrl_1_P16368     6   12      + 10.897   8e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00304          GGACGAG\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "2 oRNAment     FXR2      1.2      DGACRRR\n",
      "  motif_id motif_alt_id       sequence_name start stop strand  score p.value\n",
      "1      1.2      DGACRRR Actg2_ctrl_1_P16368     6   12      + 10.897   8e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00304          GGACGAG\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "1   mCross     ILF3    1.5.1  BKSAGCCTCNB\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score  p.value\n",
      "1    1.5.1  BKSAGCCTCNB Actg2_ctrl_1_P16368     8   18      - 14.2879 4.53e-06\n",
      "   q.value matched_sequence\n",
      "1 0.000136      CTGAGCCTCGT\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "3 oRNAment    MBNL1    1.1.2      GCWTGCN\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score  p.value\n",
      "1    1.1.2      GCWTGCN Actg2_ctrl_1_P16368    19   25      - 11.0594 5.35e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00203          GCTTGCT\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "1   mCross     NPM1      1.7  DBCTCTGDGNB\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score p.value\n",
      "1      1.7  DBCTCTGDGNB Actg2_ctrl_1_P16368    12   22      - 13.6768 1.1e-05\n",
      "   q.value matched_sequence\n",
      "1 0.000329      TGCTCTGAGCC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "7 oRNAment    PCBP2    1.4.2      NCCCCHN\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score p.value\n",
      "1    1.4.2      NCCCCHN Actg2_ctrl_1_P16368     1    7      - 10.4898   8e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00304          CCCCCAC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "6 oRNAment     RBM6      1.1      HGTCCCB\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score p.value\n",
      "1      1.1      HGTCCCB Actg2_ctrl_1_P16368     4   10      - 10.6364   8e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00304          CGTCCCC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "1   mCross    SF3A3      1.5  NBCTCTGVCNB\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score  p.value\n",
      "1      1.5  NBCTCTGVCNB Actg2_ctrl_1_P16368    12   22      - 11.1579 8.07e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00242      TGCTCTGAGCC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "2   mCross    SF3B4    1.2.1  NNCTCTGMCNB\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score  p.value\n",
      "1    1.2.1  NNCTCTGMCNB Actg2_ctrl_1_P16368    12   22      - 11.5968 5.15e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00155      TGCTCTGAGCC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "3   mCross     XPO5      1.7  NBTCTGACCHB\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score p.value\n",
      "1      1.7  NBTCTGACCHB Actg2_ctrl_1_P16368    11   21      - 11.0986 6.7e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00201      GCTCTGAGCCT\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "8   mCross     XPO5      1.8  NNTGCTCTGNN\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score  p.value\n",
      "1      1.8  NNTGCTCTGNN Actg2_ctrl_1_P16368    14   24      - 12.3306 2.67e-05\n",
      "  q.value matched_sequence\n",
      "1   8e-04      CTTGCTCTGAG\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "1   mCross    XRCC6    1.3.1  BBTGCTSTGNN\n",
      "  motif_id motif_alt_id       sequence_name start stop strand   score p.value\n",
      "1    1.3.1  BBTGCTSTGNN Actg2_ctrl_1_P16368    14   24      - 13.1579 1.5e-05\n",
      "   q.value matched_sequence\n",
      "1 0.000451      CTTGCTCTGAG\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "5   mCross     FMR1      1.2  NNACCTGYGNB\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.2  NNACCTGYGNB Actg2_ctrl_2_P7825     3   13      + 11.0921 6.65e-05\n",
      "  q.value matched_sequence\n",
      "1   0.002      CAACCTGGGTC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "4   mCross     ILF3    1.7.1  NNCCCTGGTNB\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1    1.7.1  NNCCCTGGTNB Actg2_ctrl_2_P7825     3   13      - 10.4366 8.99e-05\n",
      "  q.value matched_sequence\n",
      "1  0.0027      GACCCAGGTTG\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "3 CisBP_RNA    RBM24        1      WGWGTGD\n",
      "  motif_id motif_alt_id      sequence_name start stop strand score  p.value\n",
      "1        1      WGWGTGD Actg2_ctrl_2_P7825    16   22      +  11.2 6.51e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00247          AGTGTGA\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "4 oRNAment    RBM24      1.3      WGWGTGD\n",
      "  motif_id motif_alt_id      sequence_name start stop strand score  p.value\n",
      "1      1.3      WGWGTGD Actg2_ctrl_2_P7825    16   22      +  11.2 6.51e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00247          AGTGTGA\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "1   mCross    SUGP2      1.9  BKKTCTCAGNN\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.9  BKKTCTCAGNN Actg2_ctrl_2_P7825     9   19      + 13.5493 8.98e-06\n",
      "  q.value matched_sequence\n",
      "1 0.00027      GGGTCTCAGTG\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "2   mCross    DGCR8      1.1  BYCTAGCACYW\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.1  BYCTAGCACYW Actg2_ctrl_3_P4713     8   18      - 11.8384 3.96e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00127      GTCTAGCACCC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "3   mCross     SBDS      1.3  BWCCGSGTTBC\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.3  BWCCGSGTTBC Actg2_ctrl_3_P4713     3   13      + 11.4404 7.23e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00231      CTCCAGGGTGC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "7   mCross   SERBP1      1.6  SSTGCTTGASY\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.6  SSTGCTTGASY Actg2_ctrl_3_P4713     9   19      + 11.7429 2.45e-05\n",
      "   q.value matched_sequence\n",
      "1 0.000783      GGTGCTAGACT\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "3   mCross    SF3B4      1.4  NNCTGACTCHB\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.4  NNCTGACTCHB Actg2_ctrl_3_P4713    16   26      + 10.6935 9.24e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00296      GACTGACCCCG\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "21   mCross    SRSF1    1.2.1  NNHBCTGGASV\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1    1.2.1  NNHBCTGGASV Actg2_ctrl_3_P4713     2   12      - 11.7931 2.75e-05\n",
      "   q.value matched_sequence\n",
      "1 0.000881      CACCCTGGAGC\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "23   mCross    SRSF1    1.4.2  NNMYCTGGANN\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1    1.4.2  NNMYCTGGANN Actg2_ctrl_3_P4713     2   12      - 11.1129 7.87e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00252      CACCCTGGAGC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "2   mCross    DGCR8      1.1  BYCTAGCACYW\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.1  BYCTAGCACYW Actg2_ctrl_4_P4710    10   20      - 11.8384 3.96e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00119      GTCTAGCACCC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "3   mCross     SBDS      1.3  BWCCGSGTTBC\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.3  BWCCGSGTTBC Actg2_ctrl_4_P4710     5   15      + 11.4404 7.23e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00217      CTCCAGGGTGC\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "7   mCross     SBDS      1.1  HNGGCTMCAVN\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.1  HNGGCTMCAVN Actg2_ctrl_4_P4710     1   11      + 15.1101 1.23e-06\n",
      "   q.value matched_sequence\n",
      "1 3.68e-05      CAGGCTCCAGG\n",
      "  database RBP_name motif_id motif_alt_id\n",
      "7   mCross   SERBP1      1.6  SSTGCTTGASY\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1      1.6  SSTGCTTGASY Actg2_ctrl_4_P4710    11   21      + 11.7429 2.45e-05\n",
      "   q.value matched_sequence\n",
      "1 0.000734      GGTGCTAGACT\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "21   mCross    SRSF1    1.2.1  NNHBCTGGASV\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1    1.2.1  NNHBCTGGASV Actg2_ctrl_4_P4710     4   14      - 11.7931 2.75e-05\n",
      "   q.value matched_sequence\n",
      "1 0.000826      CACCCTGGAGC\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "23   mCross    SRSF1    1.4.2  NNMYCTGGANN\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1    1.4.2  NNMYCTGGANN Actg2_ctrl_4_P4710     4   14      - 11.1129 7.87e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00236      CACCCTGGAGC\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "17   mCross    DDX3X    1.2.2  NNCGGTRGCND\n",
      "  motif_id motif_alt_id      sequence_name start stop strand  score  p.value\n",
      "1    1.2.2  NNCGGTRGCND Actg2_ctrl_5_P2922     2   12      - 11.439 5.47e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00186      CTCTGTAGCCG\n",
      "    database RBP_name motif_id motif_alt_id\n",
      "18 CisBP_RNA    SNRPA    1.6.3      WGGAGAT\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1    1.6.3      WGGAGAT Actg2_ctrl_5_P2922    14   20      - 11.5051 6.51e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00273          AGGAGAT\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "19 oRNAment    SNRPA    1.8.1      WGGAGAT\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1    1.8.1      WGGAGAT Actg2_ctrl_5_P2922    14   20      - 11.5051 6.51e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00273          AGGAGAT\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "20    RBPDB    SNRPA    1.8.2      WGGAGAT\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1    1.8.2      WGGAGAT Actg2_ctrl_5_P2922    14   20      - 11.4949 6.51e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00273          AGGAGAT\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "14   mCross    SRSF7     1.10  NNGAGATCTTY\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1     1.10  NNGAGATCTTY Actg2_ctrl_5_P2922    10   20      - 11.8684 3.54e-05\n",
      "  q.value matched_sequence\n",
      "1  0.0012      AGGAGATCCTC\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "19   mCross    SRSF7    1.5.2  NVAGATCKTCN\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1    1.5.2  NVAGATCKTCN Actg2_ctrl_5_P2922     9   19      - 11.9868 4.14e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00141      GGAGATCCTCT\n",
      "   database RBP_name motif_id motif_alt_id\n",
      "19   mCross    UCHL5     1.10  NVAGATCKTBN\n",
      "  motif_id motif_alt_id      sequence_name start stop strand   score  p.value\n",
      "1     1.10  NVAGATCKTBN Actg2_ctrl_5_P2922     9   19      - 12.1408 3.21e-05\n",
      "  q.value matched_sequence\n",
      "1 0.00109      GGAGATCCTCT\n"
     ]
    }
   ],
   "source": [
    "sequence_name = \"Actg2_ctrl\"                                ### Change Here : Actg2_hotspot, Actg2_ctrl, Actg2_random\n",
    "base_dir <- paste0(\"/home/ylee/meme/fimo_out/\", sequence_name)\n",
    "csv_dir <- paste0(base_dir, \"/alignment_csv\")\n",
    "\n",
    "fimo_base_paths <- paste0(list.dirs(base_dir, recursive = FALSE, full.names = TRUE), \"/\")\n",
    "dir.create(csv_dir)\n",
    "\n",
    "for (fimo_base_path in fimo_base_paths){\n",
    "    motif_details <- alignment_details(rbp_list, All_RBP_motifs, fimo_base_path)    \n",
    "    \n",
    "    output_dir <- paste0(csv_dir, \"/\", basename(fimo_base_path), \".csv\")\n",
    "    write.csv(motif_details, output_dir, row.names = FALSE)   \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "0da4e3d6-493f-413c-b5d9-91a3ad507101",
   "metadata": {},
   "outputs": [],
   "source": [
    "if (sequence_name == \"Actg2_hotspot\"){\n",
    "    mapped_data <- read.csv(\"/home/ylee/meme/Actg2_sequence/Actg1_hotspot101_Actg2_mapped_sequecnes_68_percent.csv\", header = TRUE, fill = TRUE)\n",
    "} else if (sequence_name == \"Actg2_ctrl\"){\n",
    "    mapped_data <- read.csv(\"/home/ylee/meme/Actg2_sequence/Actg1_ctrl_Actg2_mapped_sequecnes_68_percent.csv\", header = TRUE, fill = TRUE)\n",
    "} else {     \n",
    "    mapped_data <- read.csv(\"/home/ylee/meme/Actg2_sequence/Actg1_random_Actg2_mapped_sequecnes_68_percent.csv\", header = TRUE, fill = TRUE)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "52eb616c-94e8-49e4-a1a8-0ca1a866f7e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "csv_paths <- list.files(csv_dir, recursive = FALSE, full.names = TRUE)\n",
    "csv_paths <- csv_paths[grepl(paste0(\"^\", sequence_name, \"_\\\\d+\\\\.csv$\"), basename(csv_paths))]\n",
    "\n",
    "results <- alignment_combined_details(csv_paths, mapped_data, sequence_name)\n",
    "write.csv(results$mapped_data, paste0(csv_dir, \"/\", sequence_name, \"_mapped_motif_alignments_summary.csv\"), row.names = FALSE) \n",
    "write.csv(results$alignment_combined, paste0(csv_dir, \"/\", sequence_name, \"_mapped_motif_alignments.csv\"), row.names = FALSE)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2cbbfe44-7057-4c4d-84e2-9b79ffbad3e4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "r_4.1.2",
   "language": "R",
   "name": "r_4.1.2"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
